packages/browser/src/plugins/remote-loader/index.ts

```diff
@@ -113,25 +123,40 @@ export class ActionDestination implements DestinationPlugin {
     return this.action.isLoaded()
   }
 
-  ready(): Promise<unknown> {
-    return this.action.ready ? this.action.ready() : Promise.resolve()
+  async ready(): Promise<boolean> {
+    try {
+      await this.loadPromise.promise
+      return true
+    } catch {
+      return false
+    }
   }
 
   async load(ctx: Context, analytics: Analytics): Promise<unknown> {
+    if (this.loadPromise.isSettled()) {
+      return this.loadPromise.promise
+    }
+
     try {
       recordIntegrationMetric(ctx, {
         integrationName: this.action.name,
         methodName: 'load',
         type: 'action',
       })
-      return await this.action.load(ctx, analytics)
+
+      const loadP = this.action.load(ctx, analytics)
+
+      this.loadPromise.resolve(await loadP)
```

Comment:
Do we want this await here, instead of just loadP? Promise<Promise<val>> unwraps to Promise<val> automatically, so should resolve to the same thing.

Looks a bit strange to await a promise result but then return the underlying promise -- I want to just make sure we aren't waiting if we don't need to.

LMK if missing something!

Reply:
Yeah I got a bit caught up by this too. I think the await is here so that if load throws an error it's catch by the catch handler and we can record the metric.

Reply:
Ah that makes sense